app/validators/personal_key_validator.rb

```diff
@@ -1,20 +1,20 @@
 module PersonalKeyValidator
   extend ActiveSupport::Concern
 
-  included do
-    validate :valid_personal_key?
-  end
-
   private
 
   def normalize_personal_key(personal_key = nil)
     return nil if personal_key.blank?
     personal_key_generator.normalize(personal_key)
   end
 
-  def valid_personal_key?
-    return false unless personal_key =~ /\A#{PersonalKeyFormatter.new.regexp}\Z/
-    personal_key_generator.verify(personal_key)
+  def check_personal_key
+    return if personal_key_format_matches? && personal_key_generator.verify(personal_key)
+    errors.add :personal_key, :personal_key_incorrect
+  end
+
+  def personal_key_format_matches?
+    personal_key =~ /\A#{PersonalKeyFormatter.new.regexp}\Z/
```

Comment:
@zachmargolis I update the `PeronalKeyFormatter` class. I split up the method and moved the variables into constants, since we shouldn't have to recompute the regex string.